instrumentation-api-appender/src/test/java/io/opentelemetry/instrumentation/api/appender/GlobalLogEmitterProviderTest.java

```diff
@@ -32,26 +34,31 @@ void testGlobalBeforeSet() {
   @Test
   void setThenSet() {
     setLogEmitterProvider();
-    assertThatThrownBy(() -> GlobalLogEmitterProvider.set(NoopLogEmitterProvider.INSTANCE))
+    assertThatThrownBy(() -> GlobalLogEmitterProvider.set(mock(LogEmitterProvider.class)))
         .isInstanceOf(IllegalStateException.class)
         .hasMessageContaining("GlobalLogEmitterProvider.set has already been called")
         .hasStackTraceContaining("setLogEmitterProvider");
   }
 
   @Test
   void getThenSet() {
-    assertThat(getLogEmitterProvider()).isInstanceOf(NoopLogEmitterProvider.class);
-    assertThatThrownBy(() -> GlobalLogEmitterProvider.set(NoopLogEmitterProvider.INSTANCE))
-        .isInstanceOf(IllegalStateException.class)
-        .hasMessageContaining("GlobalLogEmitterProvider.set has already been called")
-        .hasStackTraceContaining("getLogEmitterProvider");
+    LogEmitterProvider existingProvider = GlobalLogEmitterProvider.get();
+    assertSame(existingProvider, NoopLogEmitterProvider.INSTANCE);
+    LogEmitterProvider newProvider = mock(LogEmitterProvider.class);
+    GlobalLogEmitterProvider.set(newProvider);
+    assertSame(newProvider, GlobalLogEmitterProvider.get());
   }
 
-  private static void setLogEmitterProvider() {
+  @Test
+  void okToSetNoopMultipleTimes() {
+    GlobalLogEmitterProvider.set(NoopLogEmitterProvider.INSTANCE);
+    GlobalLogEmitterProvider.set(NoopLogEmitterProvider.INSTANCE);
+    GlobalLogEmitterProvider.set(NoopLogEmitterProvider.INSTANCE);
     GlobalLogEmitterProvider.set(NoopLogEmitterProvider.INSTANCE);
+    // pass
   }
 
-  private static LogEmitterProvider getLogEmitterProvider() {
-    return GlobalLogEmitterProvider.get();
+  private void setLogEmitterProvider() {
```

Comment:
Errorprone sure can be annoying sometimes:
>: warning: [MethodCanBeStatic] A private method that does not reference the enclosing instance can be static

```suggestion
  private static void setLogEmitterProvider() {
```